Documentation/contributing.rst

```diff
@@ -1274,76 +1540,269 @@ simplify our Jenkinsfiles. The exported methods are:
+       case ``v1.0``
+
+#. Follow standard procedures to get the 
+#. Checkout out the stable branch and pull your merged changes:
+
+   ::
+
+       git checkout v1.0; git pull
+
+#. Create a release tag:
+
+   ::
+
+       git tag -a v1.0.3 -m 'Release v1.0.3`
+
+#. Build the binaries and push it to the release bucket:
+
+   ::
+
+       DOMAIN=releases.cilium.io ./contrib/release/uploadrev v1.0.3
+
+
+   This step will print a markdown snippet which you will need when crafting
+   the GitHub release so make sure to keep it handy.
+
+   .. note:
+
+       This step requires valid AWS credentials to be available via the
+       environment variables ``AWS_ACCESS_KEY_ID`` and
+       ``AWS_SECRET_ACCESS_KEY``. Ping in the ``#development`` channel on Slack
+       if you have no access.
+
+#. Build the container images and push them
+
+   ::
+
+      DOCKER_IMAGE_TAG=v1.0.3 make docker-image
+      docker push cilium/cilium:v1.0.3
+
+   .. note:
+
+      This step requires you to login with ``docker login`` first and it will
+      require your Docker hub ID to have access to the ``Cilium`` organization.
+
+#. Push the git release tag
+
+   ::
+
+       git push --tags
+
+#. `Create a GitHub release <https://github.com/cilium/cilium/releases/new>`_:
+
+   #. Choose the correct target branch, e.g. ``v1.0``
+   #. Choose the correct target tag, e.g. ``v1.0.3``
+   #. Title: ``1.0.3``
+   #. Check the ``This is a pre-release`` box if you are releasing a release
+      candidate.
+   #. Fill in the release description:
+
+      ::
+
+           Changes
+           -------
+
+           ```
+           << contents of NEWS.rst for this release >>
+           ```
+
+           Release binaries
+           ----------------
+
+           << contents of snippet outputed by uploadrev >>
+
+   #. Preview the description and then publish the release
+
+#. Announce the release in the ``#general`` channel on Slack
+
+
+.. _minor_release_process:
+
+Minor Release Process
+~~~~~~~~~~~~~~~~~~~~~
+
+On Freeze date
+^^^^^^^^^^^^^^
+
+#. Fork a new release branch from master:
+
+   ::
+
+       git checkout master; git pull
+       git checkout -b v1.2
+       git push
+
+#. Protect the branch using the GitHub UI to disallow direct push and require
+   merging via PRs with proper reviews.
+
+#. Follow the :ref:`generic_release_process` to release ``v1.2.0-rc1``.
+
+#. Prepare the master branch for the next development cycle:
+
+   ::
+
+       git checkout master; git pull
+
+#. Update the ``VERSION`` file to contain ``v1.2.90``
+#. Add the ``VERSION`` file using ``git add`` and create & merge a PR titled
+   ``Prepare for 1.3.0 development``.
+#. Update the release branch on
+    `Jenkins <https://jenkins.cilium.io/job/cilium-ginkgo/job/cilium/>`_ to be
+    tested on every change and Nightly.
+
+
+For the final release
+^^^^^^^^^^^^^^^^^^^^^
+
+#. Follow the :ref:`generic_release_process` to create the final replace and replace
+   ``X.Y.0-rcX`` with ``X.Y.0``.
+
```

Comment:
#. Create labels: `backport-done/X.Y-1` `backport-pending/X.Y-1` `backport/X.Y-1` and `needs-backport/X.Y-1`